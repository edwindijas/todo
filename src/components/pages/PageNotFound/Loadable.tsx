import { lazy } from 'react'
import Loader from '@/components/molecules/Loader'

const PageNotFound = lazy(async () => await import('.'))

const Loadable = (): JSX.Element => (
  <Loader >
    <PageNotFound />
  </Loader>
)

export default Loadable