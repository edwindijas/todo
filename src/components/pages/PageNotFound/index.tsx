const PageNotFound = (): JSX.Element => (<div data-testid='page-not-found' className='w-full h-full flex items-center justify-center' >
    <h1 className='font-bold text-4xl' >Page Not Found</h1>
  </div>)

export default PageNotFound