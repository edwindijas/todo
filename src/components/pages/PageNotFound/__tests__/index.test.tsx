import '@testing-library/jest-dom/jest-globals';
import '@testing-library/jest-dom';
import { render, waitFor } from '@testing-library/react'
import PageNotFound from '..'

it ('Renders snapshot correctly', async () => {
  const {container, getByText} = render(<PageNotFound />)
  await waitFor( () => { expect(getByText('Page Not Found')).toBeInTheDocument() })
  expect(container).toMatchSnapshot()
})
