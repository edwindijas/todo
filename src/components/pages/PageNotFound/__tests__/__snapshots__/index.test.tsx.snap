// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders snapshot correctly 1`] = `
<div>
  <div
    class="w-full h-full flex items-center justify-center"
    data-testid="page-not-found"
  >
    <h1
      class="font-bold text-4xl"
    >
      Page Not Found
    </h1>
  </div>
</div>
`;
